autogpt/memory/vector/utils.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, overload
+
+import numpy as np
+import numpy.typing as npt
+import openai
+
+from autogpt.config import Config
+from autogpt.llm.utils import metered, retry_openai_api
+from autogpt.logs import logger
+
+Embedding = list[np.float32] | np.ndarray[Any, np.dtype[np.float32]]
+"""Embedding vector"""
+TText = list[int]
+"""Token array representing text"""
+
+
+@overload
+def get_embedding(input: str | TText) -> Embedding:
```

Comment:
Why would we add all the complexity of overloading this if I can just do:

```python
my_text = [1, 2, 3]
my_texts = [my_embedding, my_embedding]

my_embedding = get_embedding(my_text)
my_embeddings = [get_embedding(text) for text in my_texts]
```

Reply:
The API docs specifically advise to use batch requests where appropriate, so that's what I implemented.

Reply:
Makes sense.